release.py

```diff
@@ -177,18 +183,66 @@ def push() -> None:
             raise typer.Exit(result)
 
 
-def check_preconditions() -> None:
+def ensure_all_committed_files() -> Optional[List[str]]:
+    command = ("git", "ls-files", "-m")
+    typer.echo(" ".join(command))
+    result = subprocess.run(command, capture_output=True)
+
+    uncommitted_files = result.stdout.decode("UTF-8").split("\n")[:-1]
```

Comment:
```suggestion
    result = subprocess.run(command, capture_output=True, text=True)

    uncommitted_files = result.stdout.split("\n")[:-1]
```
This makes it so that you get `str` and not `bytes`